src/Game/XPBar.java

```diff
@@ -156,6 +156,10 @@ void draw(Graphics2D g, BufferedMouseClick bufferedMouseClick) {
       g.setColor(Renderer.color_fatigue);
     }
     g.fillRect(x + 1, y + 1, percent, bounds.height - 2);
+    if (percent_progress > 0.0f) {
+      g.setColor(Renderer.color_fatigue);
+      g.fillRect(x + percent + 1, y + 1, percent_progress, bounds.height - 2);
```

Comment:
Actually I noticed the XP bar fill doesn't go all the way to the end of the bar if you're > lvl 1 on a skill. I think the below code fixes the problem, though please test edge conditions like lvl 99+

```suggestion
      g.fillRect(x + percent + 1, y + 1, percent_progress + (percent > 0.0f ? 1 : 0), bounds.height - 2);
```

Reply:
Good change but for it to look a bit more clean did that calculation when setting percent_progress

Reply:
good call